src/script.js

```diff
@@ -0,0 +1,231 @@
+const sortDownButton = document.querySelector(SORT_DOWN_BTN_SELECTOR);
+const sortUpButton = document.querySelector(SORT_UP_BTN_SELECTOR);
+const sortButtonsContainer = document.querySelector(SORT_BTNS_SELECTOR);
+const pokemonImagesContainer = document.querySelector(POKEMON_IMAGES_SELECTOR);
+const pokemonCatchText = document.querySelector(POKEMON_TEXT_SELECTOR);
+
+let pendingTasks = 0;
+
+const tasks = new ItemManager();
+const pokemonClient = new PokemonClient();
+
+addTaskButton.onclick = () => {
+   renderAndAnimate();
+};
+
+taskInput.onkeypress = (e) => {
+   if (e.key === "Enter") {
+      renderAndAnimate();
+   }
+};
+
+async function renderAndAnimate() {
+   if (await addTask()) {
+      const newTask = renderTasks();
+      createTaskAnimation(newTask);
+   }
+}
+
+clearButton.onclick = () => {
+   tasks.clear();
+   renderTasks();
+   pendingTasksUpdate(0);
+};
+
+sortDownButton.onclick = () => {
+   tasks.sortDown();
+   renderTasks();
+};
+
+sortUpButton.onclick = () => {
+   tasks.sortUp();
+   renderTasks();
+};
+
+function renderTasks() {
+   let lastTaskAdded;
+   while (allTasksContainer.firstChild) {
+      allTasksContainer.removeChild(allTasksContainer.lastChild);
+   }
+   tasks.items.forEach((item) => {
+      const taskContainer = createTaskContainer();
+      const newTask = createNewTask(taskContainer, item);
+      const deleteTask = createDeleteTaskButton(taskContainer);
+      addHoverReveal(taskContainer, deleteTask, newTask);
+      lastTaskAdded = newTask;
+   });
+
+   return lastTaskAdded;
+}
+
+async function addTask() {
+   const taskUserInput = taskInput.value;
+   taskInput.value = "";
+   const allPokemonNames = await pokemonClient.getAllPokemonNames();
+   if (taskUserInput.trim().length === 0) {
+      alert("Please fill in a task");
+
+      return false;
+   } else if (
+      !isNaN(taskUserInput) ||
+      allPokemonNames.includes(taskUserInput)
+   ) {
+      // For single Pokemon entry
+      let pokemonData = await pokemonClient.getPokemon(taskUserInput);
+      let render = false;
+      render = pokemonTasksHandle(pokemonData, taskUserInput, 0);
+
+      return render;
+   } else if (
+      taskUserInput
+         .replace(/\s/g, "")
+         .split(",")
+         .every((elem) => !isNaN(elem) || allPokemonNames.includes(elem))
+   ) {
+      // For multiple Pokemons entry
+      let pokemonIDS = taskUserInput.replace(/\s/g, "").split(","); // "1, 2, 3" => [1,2,3]
+      const pokemonData = await Promise.all(
+         pokemonIDS.map((id) => pokemonClient.getPokemon(id))
+      );
+      let render = false;
+      pokemonData.forEach((pokemon, i) => {
+         render = pokemonTasksHandle(pokemon, pokemonIDS, 1, i) || render; // Render even if one pokemon is not listed in tasks
+      });
+
+      return render;
+   } else {
+      // For regular boring non-pokemon tasks
+      tasks.add(taskUserInput);
+      pendingTasksUpdate("+");
+
+      return true;
+   }
```

Comment:
this method is very complex, try to think on a way to make it simpler.
I would suggest of always thinking in plural mode - you are given with an input, try to split it by `","`, them trim each item in the array, and then threat it as an array of inputs no matter if it's given with 1 item or multiple items.

Reply:
Deleted all single pokemon handling. You're right, really no need to divide them, all of the functions for multiple pokemon works great with single one as well.